tests/test_recording.py

```diff
@@ -42,6 +42,49 @@ def test_without_network():
     assert not cassette_path.exists()
 
 
+def test_cassette_recording_rewrite(testdir):
+    testdir.makepyfile(
+        """
+        import pytest
+        import requests
+
+        @pytest.mark.vcr
+        def test_{}(httpbin):
+            assert requests.get(httpbin.url + "/uuid").status_code == 200
+
+        @pytest.mark.vcr
+        class TestSomething:
+            def test_with_network(self, httpbin):
+                assert requests.get(httpbin.url + "/uuid").status_code == 200
+    """.format(
+            string.ascii_letters
+        )
+    )
+
+    # If recording is enabled
+    result = testdir.runpytest("--record-mode=rewrite")
+    result.assert_outcomes(passed=2)
+
+    # Then tests that use network will create cassettes
+    test_function_cassette_path = testdir.tmpdir.join(
+        "cassettes/test_cassette_recording_rewrite/test_{}.yaml".format(string.ascii_letters)
+    )
+    test_function_size = test_function_cassette_path.size()
+    assert test_function_size
+    test_class_cassette_path = testdir.tmpdir.join(
+        "cassettes/test_cassette_recording_rewrite/TestSomething.test_with_network.yaml"
+    )
+    test_class_size = test_class_cassette_path.size()
+    assert test_class_size
+
+    # Second run will pass as well
+    result = testdir.runpytest("--record-mode=rewrite")
```

Comment:
What about checking that uuid values are actually different after recording? otherwise, this test will pass, if e.g. this cassette wasn't touched at all